projects/vdk-core/src/vdk/internal/builtin_plugins/run/cli_run.py

```diff
@@ -120,6 +120,26 @@ def __warn_on_python_version_disparity(
                         """
                     )
 
+    def __log_exec_result(self, execution_result):
+        # On some platforms, if the size of a string is too large, the
+        # logging module starts throwing OSError: [Errno 40] Message too long,
+        # so it is safer if we split large strings into smaller chunks.
+        string_exec_result = str(execution_result)
+        if len(string_exec_result) > 5000:
+            temp_exec_result = json.loads(string_exec_result)
+            steps = temp_exec_result.pop("steps_list")
```

Comment:
Why pop? and not get?

Reply:
Refactoring is out of scope for this PR. I'm just copying existing code to a new function. We can open a ticket to take a look at this at a later time.